src/main/java/org/cryptomator/common/vaults/Vault.java

```diff
@@ -142,12 +166,23 @@ public synchronized void unlock(MasterkeyLoader keyLoader) throws CryptoExceptio
 		if (cryptoFileSystem.get() != null) {
 			throw new IllegalStateException("Already unlocked.");
 		}
+		var fallbackProvider = mountProviders.stream().findFirst().orElse(null);
+		var defMntServ = ObservableUtil.mapWithDefault(settings.mountService, serviceName -> mountProviders.stream().filter(s -> s.getClass().getName().equals(serviceName)).findFirst().orElse(fallbackProvider), fallbackProvider).getValue();
+		var selMntServ = ObservableUtil.mapWithDefault(vaultSettings.mountService, serviceName -> mountProviders.stream().filter(s -> s.getClass().getName().equals(serviceName)).findFirst().orElse(defMntServ), defMntServ);
+		var fuseRestartRequired = selMntServ.map(s -> //
+				firstUsedProblematicFuseMountService.get() != null //
+						&& VaultModule.isProblematicFuseService(s) //
+						&& !firstUsedProblematicFuseMountService.get().equals(s)).getValue();
```

Comment:
Please deduplicate the lines here by creating a new static method in the mount/Mounter class, because you use the same lines also at three other spots. Also, check at each spot, if you actually need the observability or not. In the latter case, remove the observeable wrappers.